### PR TITLE
Refactor FlavorAssigner

### PR DESCRIPTION
**Pull request by gabesaba**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Simplifies FlavorAssigner in preparation for #3405

#### Does this PR introduce a user-facing change?
```release-note
Change, and in some scenarios fix, the status message displayed to user when a workload doesn't fit in available capacity.
```

### CODE REVIEW

**Comment by gabesaba** [comment]
> /lgtm

thanks for the review!

File: pkg/scheduler/flavorassigner/flavorassigner.go [hunk 1]
```diff
@@ -603,47 +603,37 @@ func flavorSelector(spec *corev1.PodSpec, allowedKeys sets.Set[string]) nodeaffi
...
+		if a.oracle.IsReclaimPossible(log, a.cq, *a.wl, fr, val) {
+			mode = reclaim
 		}
+	} else if a.canPreemptWhileBorrowing() {
+		mode = preempt
+		borrow = true
 	}
-	status.append(msg)
+
+	status.append(fmt.Sprintf("insufficient unused quota for %s in flavor %s, %s more needed",
```
Comment (PBundyra) [inline pkg/scheduler/flavorassigner/flavorassigner.go:634]: Do we want to set this message even if `val <= rQuota.Nominal`?

Comment (gabesaba) [reply pkg/scheduler/flavorassigner/flavorassigner.go:634]: We did this (a similar message) before, and I think the message is still relevant/accurate - other quota is being used

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve
We will be cherry-picking this one probably too.

**Comment by mimowo** [comment]
/release-note-edit
```release-note
Change, and in some scenarios fix, the status message displayed to user when a workload doesn't fit in available capacity.
```

**Comment by mimowo** [comment]
/kind bug 
Due to the inaccurate message as per https://github.com/kubernetes-sigs/kueue/pull/3536#discussion_r1842569501.

**Pull request merged**